hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllBackupOperation.java

```diff
@@ -112,25 +131,43 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
             Records.writeExpiry(out, expiryMetadata);
         }
         out.writeBoolean(disableWanReplicationEvent);
+
+        if (noWanReplicationKeys == null) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeByteArray(noWanReplicationKeys.toByteArray());
+        }
     }
 
     @Override
+    @SuppressWarnings("checkstyle:magicnumber")
     protected void readInternal(ObjectDataInput in) throws IOException {
         super.readInternal(in);
 
         int size = in.readInt();
-        List keyRecordExpiry = new ArrayList<>(size * 3);
+        List keyRecordExpiryWan = new ArrayList<>(size * 3);
         for (int i = 0; i < size; i++) {
             Data dataKey = IOUtil.readData(in);
             Record record = Records.readRecord(in);
             ExpiryMetadata expiryMetadata = Records.readExpiry(in);
 
-            keyRecordExpiry.add(dataKey);
-            keyRecordExpiry.add(record);
-            keyRecordExpiry.add(expiryMetadata);
+            keyRecordExpiryWan.add(dataKey);
+            keyRecordExpiryWan.add(record);
+            keyRecordExpiryWan.add(expiryMetadata);
         }
-        this.keyRecordExpiry = keyRecordExpiry;
+        this.keyRecordExpiry = keyRecordExpiryWan;
         this.disableWanReplicationEvent = in.readBoolean();
+        this.noWanReplicationKeys = null;
+
+        // RU_COMPAT_5_3
+        try {
```

Comment:
As discussed on slack, no `EOFException` will be thrown here, because the backup operations are contained within a `Backup` operation which involves serde of additional data after the end of the `PutAllBackupOperation` itself.
So, instead, we should rely on gating the serde of `noWanReplicationKeys` on `out.getVersion()` / `in.getVersion()`

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/24928/commits/12f00318cd7d78ba56a65173931efd2a5d621580